src/content/docs/en/guides/cms/hashnode.mdx

```diff
@@ -0,0 +1,375 @@
+        publication(host: "${myHashnodeURL}") {
+          post(slug: $slug) {
+            author{
+              name
+              profilePicture
+            }
+            publishedAt
+            title
+            subtitle
+            readTimeInMinutes
+            content{
+              html
+            }
+            tags {
+              name
+              slug
+            }
+            coverImage {
+              url
+            }
+          }
+        }
+      }
+    `,
+    { slug: slug }
+  );
+      
+  return data.publication.post;
+};
+
+```
+
+And setup the schema file to define the shape of the data returned from the Hashnode API.
+
+```ts title="src/lib/schema.ts"
+
+import { z } from "astro/zod";
+
+export const PostSchema = z.object({
+    author: z.object({
+        name: z.string(),
+        profilePicture: z.string(),
+        }),
+    publishedAt: z.string(),
+    title: z.string(),
+    subtitle: z.string(),
+    brief: z.string(),
+    slug: z.string(),
+    readTimeInMinutes: z.number(),
+    content: z.object({
+        html: z.string(),
+    }),
+    tags: z.array(z.object({
+        name: z.string(),
+        slug: z.string(),
+    })),
+    coverImage: z.object({
+        url: z.string(),
+    }),
+})
+
+export const AllPostsDataSchema = z.object({
+    publication: z.object({
+        title: z.string(),
+        posts: z.object({
+            pageInfo: z.object({
+                hasNextPage: z.boolean(),
+                endCursor: z.string(),
+            }),
+            edges: z.array(z.object({
+                node: PostSchema,
+            })),
+        }),
+    }),
+})
+
+export const PostDataSchema = z.object({
+    publication: z.object({
+        title: z.string(),
+        post: PostSchema,
+    }),
+})
+
+export type Post = z.infer<typeof PostSchema>
+export type AllPostsData = z.infer<typeof AllPostsDataSchema>
+export type PostData = z.infer<typeof PostDataSchema>
+
+```
+
+### Displaying a list of posts
+
+The page `src/pages/index.astro` will display a list of posts, each with a description and link to its own page.
+
+<FileTree title="Project Structure">
+- src/
+ - lib/
+    - client.ts
+    - schema.ts
+  - pages/
+    - **index.astro**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+Import `getAllPosts()` in the Astro frontmatter to fetch the data from Hashnode.
+
+```astro title="src/pages/index.astro"
+---
+import { getAllPosts } from '../lib/client';
+
+const data = await getAllPosts();
+const allPosts = data.publication.posts.edges;
+
+---
+```
+
+Fetching via the graphql client returns an array of objects containing the properties for each post such as:
+- `title` - the title of the post
+- `brief` - the HTML rendering of the content of the post
+- `coverImage.url` - the source URL of the featured image of the post
+- `slug` - the slug of the post
+
+Use the `posts` array returned from the fetch to display a list of blog posts on the page.
```

Comment:
``````suggestion
Fetching via `getAllPosts()` returns an array of objects containing the properties for each post such as:
- `title` - the title of the post
- `brief` - the HTML rendering of the content of the post
- `coverImage.url` - the source URL of the featured image of the post
- `slug` - the slug of the post

Use the `posts` array returned from the fetch to display a list of blog posts on the page.
``````